prefect_saturn/core.py

```diff
@@ -136,91 +148,131 @@ def register_flow_with_saturn(self, flow: Flow) -> bool:
-        #       to load the flow. Those variables will be overridden by
-        #       Kubernetes in all the places where it matters, and values
-        #       set in the image are overridden by those from kubernetes
-        #       https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container
-        flow.storage = Docker(
-            base_image=self._saturn_base_image,
-            image_name=storage_details["image_name"],
-            image_tag=image_tag,
-            registry_url=storage_details["registry_url"],
-            prefect_directory="/tmp",
-            env_vars={"SATURN_TOKEN": "placeholder-token", "BASE_URL": "placeholder-url"},
-            python_dependencies=["kubernetes"],
-            ignore_healthchecks=True,
+        Given a flow, set up all the details needed to run it on
+        a Saturn Dask cluster.
+
+        This method modifies the following components of ``Flow`` objects
+        passed to it.
+
+        * ``.storage``: a ``Webhook`` storage instance is added
+        * ``.environment``: a ``KubernetesJobEnvironment`` with a ``DaskExecutor``
+            is added. This environment will use the same image as the notebook
+            from which this code is run.
+        """
+        self._set_flow_metadata(flow)
+
+        storage = self._get_storage()
+        flow.storage = storage
+
+        environment = self._get_environment(
+            cluster_kwargs=dask_cluster_kwargs, adapt_kwargs=dask_adapt_kwargs
         )
+        flow.environment = environment
+
+        flow.storage.add_flow(flow)
+
+        # tell Saturn to store the flow
+        self._store_flow(flow=flow)
         return flow
 
-    def build_storage(self, flow: Flow) -> Response:
+    def _get_storage(self) -> Webhook:
         """
-        Actually build and push the storage
+        Create a `Webhook` storage object with Saturn-y details.
         """
-        if self._saturn_flow_id is None:
-            raise RuntimeError(Errors.NOT_REGISTERED)
+        storage = Webhook(
+            build_request_kwargs={
+                "url": "${BASE_URL}api/prefect_cloud/flows/" + self.flow_id + "/flow_content",
+                "headers": {
+                    "Content-Type": "application/octet-stream",
+                    "Authorization": "token ${SATURN_TOKEN}",
+                },
+            },
+            build_request_http_method="POST",
+            get_flow_request_kwargs={
+                "url": "${BASE_URL}api/prefect_cloud/flows/"
+                + self.flow_id
+                + "/flow_content?flow_version_id="
+                + self.flow_version_id,
+                "headers": {
+                    "Accept": "application/octet-stream",
+                    "Authorization": "token ${SATURN_TOKEN}",
+                },
+            },
+            get_flow_request_http_method="GET",
+        )
+
+        return storage
+
+    def _store_flow(self, flow: Flow) -> Response:
```

Comment:
@jameslamb you mentioned this could possibly be removed.

Reply:
yeah! Thanks for the nudge on `build=FALSE`. Trying this right now, should just take a few minutes to check.

Reply:
I was able to eliminate `_store_flow()`. Really glad you asked! It was redundant with how `Webhook` storage already works.

That simplifies things a bit.